### PR TITLE
Update scipy.org links following the mailing-list migration

### PR DESCRIPTION
**Pull request by saimn**
Not sure if the mailing-list will stay on python.org, but it does not hurt to update these links.

### CODE REVIEW

**Comment by saimn** [comment]
@astrobot - You rock !

File: docs/credits.rst [hunk 1]
```diff
@@ -227,7 +227,7 @@ Other Credits
 * Andrew Pontzen and the `pynbody <https://github.com/pynbody/pynbody>`_ team
   (For code that grew into :mod:`astropy.units`)
 * Everyone on `astropy-dev <http://groups.google.com/group/astropy-dev>`_
-  and the `astropy mailing list <http://mail.scipy.org/mailman/listinfo/astropy>`_
+  and the `astropy mailing list <https://mail.python.org/mailman/listinfo/astropy>`_
```
Comment (bsipocz) [inline docs/credits.rst]: Sphinx complains about duplicate target name https://travis-ci.org/astropy/astropy/jobs/217620466#L3377. Could you please use the same url (http --> https) everywhere (e.g. in docs/index.rst)? I think that should solve the issue.

Comment (saimn) [inline docs/credits.rst]: @bsipocz - Good catch, thanks. The mailing lists links are defined in the `conf.py`, so I modified all references to use these instead.

**Comment by bsipocz** [comment]
This passing now (failures are unrelated), so merging.

Thanks @saimn!

**Pull request merged**